mlflow/pyfunc/model.py

```diff
@@ -226,13 +226,29 @@ class ChatModel(PythonModel, metaclass=ABCMeta):
 
     See the documentation of the ``predict()`` method below for details on that parameters and
     outputs that are expected by the ``ChatModel`` API.
+
+    Warning: Before or during the MLflow 2.20 release, we will be requiring a predict_stream
+    implementation, renaming ChatRequest to ChatCompletionRequest, and renaming ChatResponse
+    to ChatCompletionResponse
     """
 
+    def __new__(cls, *args, **kwargs):
+        _logger.warning(
+            "Before or during the MLflow 2.20 release, we will be requiring a "
+            "predict_stream implementation, renaming ChatRequest to "
+            "ChatCompletionRequest, and renaming ChatResponse to "
+            "ChatCompletionResponse"
+        )
+        return super().__new__(cls)
+
     @abstractmethod
     def predict(self, context, messages: list[ChatMessage], params: ChatParams) -> ChatResponse:
         """
         Evaluates a chat input and produces a chat output.
 
+        Warning: Before or during the MLflow 2.20 release, we will be renaming ChatResponse to
```

Comment:
We might not need to say that we're renaming; just that the output type is changing. Let's enclose in a warning .rst block as well